scripts/app-stream/start-image-builder.js

```diff
@@ -0,0 +1,97 @@
+
+const { AppStreamClient, DescribeImageBuildersCommand, CreateImageBuilderCommand } = require("@aws-sdk/client-appstream");
+const { DescribeVpcsCommand, EC2Client, DescribeSubnetsCommand} = require("@aws-sdk/client-ec2");
+
+class StartImageBuilder {
+    constructor(profile, region) {
+        console.log(`Starting Image Builder using AWS Profile ${profile} in region ${region}`);
+        this.appStreamClient = new AppStreamClient({ region, profile});
+        this.region = region;
+        this.ec2Client = new EC2Client({region, profile});
+        this.imageBuilderName = `SWBImageBuilder-${Date.now()}`;
+    }
+
+    async run() {
+        try {
+            await this.createImageBuilder();
+            await this.waitForImageBuilderToBeReady();
+
+            const imageBuilderUrl = `https://console.aws.amazon.com/appstream2/home?region=${this.region}#/images?bottomTab=details&topTab=image-builders`
+
+            console.log(`You can find your new Image Builder at this address ${imageBuilderUrl}`)
+        } catch (e) {
+            process.exit(1);
+        }
+    }
+
+    async createImageBuilder() {
+        console.log(`Starting image builder with name ${this.imageBuilderName}`)
+        try {
+            const vpcsResponse = await this.ec2Client.send(
+                new DescribeVpcsCommand({
+                    Filters: [{
+                        Name: 'isDefault',
+                        Values: [true]
+                    }]
+                })
+            );
+            const vpcId = vpcsResponse.Vpcs[0].VpcId;
+            const subnetsResponse = await this.ec2Client.send(
+                new DescribeSubnetsCommand({
+                    Filters: [
+                        {
+                            Name: 'vpc-id',
+                            Values: [vpcId]
+                        }
+                    ]
+                })
+            )
+
+            const subnetId = subnetsResponse.Subnets[0].SubnetId;
+
+            await this.appStreamClient.send(
+                new CreateImageBuilderCommand({
+                        InstanceType: 'stream.graphics.g4dn.xlarge',
+                        Name: this.imageBuilderName,
+                        ImageName: 'AppStream-Graphics-G4dn-WinServer2019-06-01-2021',
+                        DisplayName: this.imageBuilderName,
```

Comment:
I think instanceType and imageName should come in as a parameter that admin has control over. Graphics instance are more expensive and customers might not even need them since AppStream instances won't itself be used for heavy compute. Note that whatever image you use here dictates what type of instances can be used when admins setup their Fleet. You can add a link to instanceType and imageName documentation in README for admins.

Reply:
Added